// definitely-typed-requirejs/require-tests.js
/// <reference path="require.d.ts" />
define(["require", "exports", 'module'], function (require, exports, module) {
    "use strict";
    // this test does not actually reference amd module 'main.ts', create one yourself.
    require.config({
        baseUrl: '../Definitions',
        // Requires versions afaik
        paths: {
            'jquery': '../Definitions/jquery',
            'underscore': '../Definitions/underscore',
            'backbone': '../Definitions/backbone'
        },
        shim: {
            jquery: {
                exports: '$'
            },
            underscore: {
                exports: '_'
            },
            backbone: {
                deps: ['underscore', 'jquery'],
                exports: 'Backbone'
            }
        }
    });
    // load AMD module main.ts (compiled to main.js)
    // and include shims $, _, Backbone
    require(['main'], function (main, $, _, Backbone) {
        var app = main.AppMain();
        app.run();
    });
    var recOne = require.config({ baseUrl: 'js' });
    recOne(['core'], function (core) { });
    var moduleConfig = module.config();
    var moduleId = module.id;
    var moduleUri = module.uri;
});
